clippy_lints/src/operators/arithmetic_side_effects.rs

```diff
@@ -178,21 +210,31 @@ impl<'tcx> LateLintPass<'tcx> for ArithmeticSideEffects {
     }
 }
 
-/// Tells if an expression is a integer declared by value or by reference.
+/// Tells if an expression is declared by value or by reference.
 ///
-/// If `LiteralIntegerTy::Ref`, then the contained value will be `hir::ExprKind::Lit` rather
-/// than `hirExprKind::Addr`.
-enum LiteralIntegerTy<'expr, 'tcx> {
+/// If `OwnershipTy::Ref`, then the contained value will be the expression followed by `&`.
+#[derive(Clone, Copy)]
+enum ExprOwnershipTy<'expr, 'tcx> {
```

Comment:
Looks like this can be replaced with `clippy_utils::peel_hir_expr_refs`.

Reply:
It it necessary to discriminate expressions with and without references because the compiler does not warn stuff like `let _ = &200u8 + &200u8`.

Although `peel_hir_expr_refs` does remove references, it doesn't tell if the given expression has references at all.

Reply:
It both removes references and returns the number of removed references.

Reply:
Eeerrrrr... You are right, sorry.

Changed as requested.